packages/rich-text/src/component/index.js

```diff
@@ -947,29 +956,22 @@ class RichText extends Component {
 
 	Editable( props ) {
 		const {
-			tagName: Tagname = 'div',
-			style,
+			tagName: TagName = 'div',
+			style = {},
 			className,
 			placeholder,
 			forwardedRef,
 		} = this.props;
-		// Generating a key that includes `tagName` ensures that if the tag
-		// changes, we replace the relevant element. This is needed because we
-		// prevent Editable component updates.
-		const key = Tagname;
+		const ariaProps = pickBy( this.props, ( value, key ) =>
+			startsWith( key, 'aria-' ) );
 
 		return (
-			<Editable
+			<TagName
 				{ ...props }
+				{ ...ariaProps }
 				ref={ forwardedRef }
-				tagName={ Tagname }
-				style={ style }
-				record={ this.record }
-				valueToEditableHTML={ this.valueToEditableHTML }
-				aria-label={ placeholder }
-				{ ...pickAriaProps( this.props ) }
+				style={ { ...style, whiteSpace } }
```

Comment:
> Can we apply this style via CSS ? It avoids the whole need to merge the objects at all.

Oh you mean the white space rule? Yes, I considered using a stylesheet, but then this would be the only rule in the stylesheet, so I picked adding it to the style object instead of creating a whole new stylesheet for this package.

I'll adjust to check the style prop.

Reply:
I wonder if React has some internal optimisations for the `style` prop on an element. It's a bit strange to expect React users to optimise these style objects that are so commonly used.

Reply:
> I wonder if React has some internal optimisations for the `style` prop on an element. It's a bit strange to expect React users to optimise these style objects that are so commonly used.

Yeah, I'm not sure if they do, but as a general principle this is also why I tend to avoid objects as props in the first place. `ariaProps` is another good example of where this becomes needlessly difficult to manage. In retrospect, I'm not sure why we didn't just expect people to pass the specific prop they wanted to apply (`aria-foo`). I seem to recall some limitation that led us down that path, but certainly made things more difficult.

Reply:
Right, I try to avoid them as much as possible too. Worth noting that `ariaProps` is picked from `props` though, there's no `ariaProps` prop. The picking seems to be there to avoid passing all `RichText` props to the element.

Reply:
> Right, I try to avoid them as much as possible too. Worth noting that `ariaProps` is picked from `props` though, there's no `ariaProps` prop. The picking seems to be there to avoid passing all `RichText` props to the element.

Oh! You're right, I definitely mis-recalled the behavior.

So maybe the opposite is true, that we have headaches because we chose to pick props than to accept an object 😄 Maybe in the future we could just be passing through all the props, so we don't need to be doing the explicit picking?

Reply:
I think we either need to pick the props we want to pass on, or delete the props we don't want to pass on. Both of these are explicit in a way I guess, but I'm fine with trying to delete the props we don't want to pass on. :) Either way there will be some sort of picking. We can't pass any unknown props to an element.